### PR TITLE
Implement turn uris received callback

### PR DESCRIPTION
**Pull request by dylonChime**
### Description of changes:
Implement callback that allows modification of TURN uris by builder.
Pass signaling url into video client start since requestTurnCreds will only be called as a backup.
### Testing done:
Joined a meeting with 2 participants and ensured video was sent and received.
#### Manual test cases (add more as needed):

- [x ] Join meeting without CallKit
- [ ] Join meeting with CallKit as Incoming
- [ ] Join meeting with CallKit as Outgoing
- [x ] Leave meeting
- [ ] Rejoin meeting
- [ ] See metrics
- [x ] See attendee presence
- [x ] Send audio
- [x ] Receive audio
- [x ] Mute self
- [x ] Unmute self
- [x ] See local mute indicator when muted
- [x ] See remote mute indicator when other is muted
- [x ] Enable and disable local video
- [x ] Enable and disable remote video from remote side
- [x ] Send local video
- [x ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Receive remote screen share

#### Integration validation (required before release)

- [x ] Unit tests pass
- [x ] Build SDK against simulator with release options
- [x ] Build SDK against device with release options
- [x ] Build SDK against simulator with debug options
- [x ] Build SDK against device with debug options
- [x ] Test Demo against simulator with SDK (debug build) in workspace
- [x ] Test Demo against device with SDK (debug build) in workspace
- [x ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [x ] Test DemoObjC against device with SDK (debug build) in workspace
- [x ] Test Demo against simulator with SDK.framework (release build)
- [x ] Test Demo against device with SDK.framework (release build)
- [x ] Test DemoObjC against simulator with SDK.framework (release build)
- [x ] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift [hunk 1]
```diff
@@ -339,6 +340,15 @@ extension DefaultVideoClientController: VideoClientDelegate {
             }
         }
     }
+    
+    public func videoClientTurnURIsReceived(_ uris: [String]?) -> [String]? {
```
Comment (tianyux-amz) [inline AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift:344]: I didn't see this in [`VideoClientDelegate`](https://code.amazon.com/packages/ChimeTincan/blobs/b9c4862a9e845366cb23186033fa52bb9db71f94/--/media_client/apple/MediaClient/VideoClient/VideoClient/VideoClientSDK.h#L125). Am I missing something?

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift [hunk 2]
```diff
@@ -339,6 +340,15 @@ extension DefaultVideoClientController: VideoClientDelegate {
...
+        guard let uris = uris else { return nil }
+        var modifiedUris = [String]()
+        for uri in uris {
+            modifiedUris.append(self.configuration.urlRewriter(uri))
+        }
+        return modifiedUris
```
Comment (tianyux-amz) [inline AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift:350]: You can directly do ```return uris?.map(self.configuration.urlRewriter)```

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift (hunk 1 above)
Comment (dylonChime) [reply AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift]: It is part of the associated media layer change.

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift [hunk 3]
```diff
@@ -339,6 +340,10 @@ extension DefaultVideoClientController: VideoClientDelegate {
             }
         }
     }
+    
+    public func videoClientTurnURIsReceived(_ uris: [String]) -> [String] {
+        return uris.map{ self.configuration.urlRewriter($0) }
```
Comment (tianyux-amz) [inline AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift:345]: Nit: Add a space before `{`: `return uris.map { self.configuration.urlRewriter($0) }`; or simply pass in `urlRewriter` as a param: `uris?.map(self.configuration.urlRewriter)`

**Review by tianyux-amz — DISMISSED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -39,6 +39,7 @@
 * Added `CameraCaptureSource`, `CaptureSourceError`, `CaptureSourceObserver`, `VideoCaptureFormat`, and `VideoCaptureSource` interfaces and enums to facilitate releasing capturers as part of the SDK.
 * Added `DefaultCameraCaptureSource` implementation of `CameraCaptureSource`.
 * Added `listVideoDevices` and `listSupportedVideoCaptureFormats` to `MediaDevice.Companion`.
+* Added TURN uris received callback.
```
Comment (zhinang-amazon) [inline CHANGELOG.md:42]: I think builder can use more information on this TURN request improvements

Comment (zhinang-amazon) [reply CHANGELOG.md:43]: I dont think this line is needed any more?

Comment (dylonChime) [reply CHANGELOG.md:49]: Since we're adding a callback, I think it still makes sense to mention it here.

**Review by zhinang-amazon — APPROVED** [review]

**Review by hokyungh — APPROVED** [review]

**Pull request merged**